lib/arrays.ts

```diff
@@ -104,3 +106,41 @@ export function map<T, U>(values: T[] | undefined, conversion: (value: T) => U):
 export function toArray<T>(value: T | T[], conversion: (valueToConvert: T) => T[] = (valueToConvert: T) => [valueToConvert]): T[] {
   return value instanceof Array ? value : conversion(value);
 }
+
+/**
+ * Get the index of the first value that matches the provided condition. -1 will be returned if no
+ * matching index is found.
+ * @param values The values to look through.
+ * @param condition The condition to check against each of the elements.
+ * @returns The first index that matches the provided condition or -1 if the condition is never
+ * satisfied.
+ */
+export function indexOf<T>(values: T[] | undefined, condition: (value: T, index: number) => boolean): number {
```

Comment:
Why do we need to re-implement that?

Reply:
The `Array.indexOf()` function only seemed to accept a value as the condition and I needed it to use a function.